code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -793,6 +803,34 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 	if (!isnull(threat_log))
 		log_threat(-cost, threat_log, reason)
 
+/// Checks if a high impact ruleset was executed and isn't dead. Only one will run at a time in most circumstances.
+/datum/game_mode/dynamic/proc/high_impact_ruleset_active()
+	. = FALSE
+	for(var/datum/dynamic_ruleset/ruleset in executed_rules)
+		if(!CHECK_BITFIELD(ruleset.flags, HIGH_IMPACT_RULESET))
+			continue
+		if(!high_impact_major_event_occured && ruleset.is_dead())
+			continue
```

Comment:
there should be some type of log that says when a ruleset has died. Might even be good to message admins if they want to bus something